spec/classes/application_spec.rb

```diff
@@ -16,11 +16,24 @@ class { 'katello::params':
 
       context 'with default parameters' do
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to create_package('tfm-rubygem-katello') }
-        it { is_expected.not_to create_package('tfm-rubygem-katello').that_requires('Anchor[katello::candlepin]') }
-        it { is_expected.to create_package('rh-postgresql12-postgresql-evr') }
-        it { is_expected.to contain_class('certs::qpid') }
-        it { is_expected.to contain_class('katello::qpid_client') }
+
+        if facts[:operatingsystemmajrelease] == '7'
```

Comment:
Is this the right way to scope the testing? I saw it used in other modules so I borrowed it throughout. @mmoll @ekohl

Reply:
In general yes, however in this specific case (if you have the time), I feel having something like this will be a tad better:
(pseudo-code)
```
if facts[:operatingsystemmajrelease] == '7'
  rubygem_prefix = 'tfm-'
  postgres_prefix = 'rh-postgresql12-'
else
  rubygem_prefix = ''
  postgres_prefix = ''
end
[...]
          it { is_expected.to create_package("#{rubygem_prefix}rubygem-katello") }
          it { is_expected.not_to create_package("#{rubygem_prefix}rubygem-katello").that_requires('Anchor[katello::candlepin]') }
          it { is_expected.to create_package("{postgres_prefix}postgresql-evr") }
```

Reply:
Given this will get ripped out in less than 3 months, would you still want this level of generic support?

Reply:
OK, nope :)